### PR TITLE
Remove files in pkg/util

### PR DESCRIPTION
**Pull request by pgimalac**
**I would advise reviewing commit by commit.** Each commit basically moves one file.

### What does this PR do?
Remove every file in `pkg/util`.

### Motivation
https://github.com/DataDog/datadog-agent/pull/32174 has shown that `pkg/util` is a mess and a bad concept, in particular considering the way Go's dead code elimination works. 

### Describe how you validated your changes

There should be no functional change, just moving things around and renaming.
CI should be enough.

### Additional Notes

I created new packages and moved some functions to existing packages when it made sense.
I also deleted functions when they were actually unused.

### CODE REVIEW

**Review by FlorianVeaux — APPROVED** [review]

**Review by guyarb — APPROVED** [review]

File: pkg/jmxfetch/utils.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import "github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+
+// GetJSONSerializableMap returns a JSON serializable map from a raw map
+func GetJSONSerializableMap(m interface{}) interface{} {
+	switch x := m.(type) {
+	// unbelievably I cannot collapse this into the next (identical) case
+	case map[interface{}]interface{}:
+		j := integration.JSONMap{}
+		for k, v := range x {
+			j[k.(string)] = GetJSONSerializableMap(v)
```
Comment (louis-cqrl) [inline pkg/jmxfetch/utils.go:19]: are we sure that in every cases `k` is a string ?

File: pkg/jmxfetch/utils.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build jmx
+
+package jmxfetch
+
+import "github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+
+// GetJSONSerializableMap returns a JSON serializable map from a raw map
```
Comment (louis-cqrl) [inline pkg/jmxfetch/utils.go:12]: is the content of the interface returned tested in other function ? If no could we add a test on it ?

**Review by vickenty — APPROVED** [review]

File: pkg/jmxfetch/utils.go (hunk 1 above)
Comment (pgimalac) [reply pkg/jmxfetch/utils.go:19]: I'm not changing anything in this PR, just moving code around

File: pkg/jmxfetch/utils.go (hunk 2 above)
Comment (pgimalac) [reply pkg/jmxfetch/utils.go:12]: I'm not changing anything in this PR, just moving code around

**Review by louis-cqrl — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Review by NouemanKHAL — APPROVED** [review]
LGTM for @DataDog/platform-integrations owned files. (`pkg/util/cloudproviders/cloudfoundry/*`)

**Review by wdhif — APPROVED** [review]
LGTM for @DataDog/container-platform files

File: pkg/util/tags/static_tags.go
Comment (clamoriniere) [inline pkg/util/tags/static_tags.go:1]: for the new package `github.com/DataDog/pkg/util/tags` could you update the `CODEOWNERS` file

with
```
/pkg/util/tags/                 @DataDog/container-platform
```

🙇

**Review by clamoriniere — COMMENTED** [review]
The change looks good and it adds clarity. 

IMO the PR should also update the CODEOWNERS file to define owner of the new packages.

File: pkg/util/tags/static_tags.go
Comment (pgimalac) [reply pkg/util/tags/static_tags.go:1]: Sure !

**Review by clamoriniere — APPROVED** [review]

**Pull request merged**